server/src/main/java/io/crate/analyze/AnalyzedColumnDefinition.java

```diff
@@ -234,6 +235,10 @@ public void analyzer(T analyzer) {
         this.analyzer = analyzer;
     }
 
+    public String analyzer() {
+        return DataTypes.STRING.sanitizeValue(analyzer);
+    }
+
```

Comment:
Could we change the type of `analyzer` to a `String` instead? If it were `T` (an `Expression` or `Symbol`) this might not do the right thing.

Reply:
It's indeed a `Symbol` -> `Literal` in `TableElementsAnalyzer.setAnalyzer` before we do evaluate but if we change `analyzer` to `String` we would have to change the workflow analyze -> evaluate_in_plan to immediate evaluate and looks we don't have all objects to create eval function at such early stage.
```
 Function<? super Symbol, Object> eval = x -> SymbolEvaluator.evaluate(
            txnCtx,
            nodeCtx,
            x,
            params,
            subQueryResults
        );
```

In our case we evaluate literals on plan creation right before we create References out of definition (https://github.com/crate/crate/pull/13238/files#diff-b3e9bec0a4a8f540cf3dec4580badee7db16c5093337574518dd3169c10d3647R87)

Reply:
Could we change the analyzer getter to return `T` instead of `String` and then let the consumer do the conversion? `sanitizeValue` is the wrong thing to use to convert expressions or symbols to strings

Reply:
I pushed a quick fixup to ensure that value is evaluated and drop usage of `sanitizeValue`  in https://github.com/crate/crate/pull/13238/commits/ff2afd97bd21b1621a0063d316419a7e1cf671f1.

Tried some other ways but looks like to tackle this properly we need to overhaul `AnalyzedTableElements/AnalyzedColumnDefinition`